Frontend test failed once because of timeout, I've restarted the test.

<details>
<summary>Error log:</summary>
Collecting pyyaml==5.1.2
  Downloading https://files.pythonhosted.org/packages/e3/e8/b3212641ee2718d556df0f23f78de8303f068fe29cdaa7a91018849582fe/PyYAML-5.1.2.tar.gz (265kB)
    100% |████████████████████████████████| 266kB 3.6MB/s 
Building wheels for collected packages: pyyaml
  Running setup.py bdist_wheel for pyyaml ... - \ | done
  Stored in directory: /home/circleci/.cache/pip/wheels/d9/45/dd/65f0b38450c47cf7e5312883deb97d065e030c5cca0a365030
Successfully built pyyaml
Installing collected packages: pyyaml
Successfully installed pyyaml-5.1.2
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting future==0.17.1
  Downloading https://files.pythonhosted.org/packages/90/52/e20466b85000a181e1e144fd8305caf2cf475e2f9674e797b222f8105f5f/future-0.17.1.tar.gz (829kB)
    100% |████████████████████████████████| 829kB 1.6MB/s 
Building wheels for collected packages: future
  Running setup.py bdist_wheel for future ... - \ | done
  Stored in directory: /home/circleci/.cache/pip/wheels/0c/61/d2/d6b7317325828fbb39ee6ad559dbe4664d0896da4721bf379e
Successfully built future
Installing collected packages: future
Successfully installed future-0.17.1
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if node.js is installed in /home/circleci/oppia/../oppia_tools
Installing Node.js
Checking if yarn is installed in /home/circleci/oppia/../oppia_tools
Removing package-lock.json
Installing yarn

WARNING: Please note that Oppia uses Yarn to manage node packages

do *NOT* use npm. For more information on how to use yarn,

visit https://yarnpkg.com/en/docs/usage.

Environment setup completed.
Checking whether Google App Engine is installed in /home/circleci/oppia/../oppia_tools/google_appengine_1.9.67/google_appengine
Downloading Google App Engine (this may take a little while)...
Download complete. Installing Google App Engine...
Checking whether google-cloud-sdk is installed in /home/circleci/oppia/../oppia_tools/google-cloud-sdk-251.0.0/google-cloud-sdk
Downloading Google Cloud SDK (this may take a little while)...
Download complete. Installing Google Cloud SDK...
Checking if pylint is installed in /home/circleci/oppia/../oppia_tools
Installing pylint
Checking if pip is installed on the local machine
Collecting pylint==1.9.4
  Downloading https://files.pythonhosted.org/packages/59/65/adcffa71fe942313c4d9e9284565d9a9e67798f4771f464e1d5dd58fea88/pylint-1.9.4-py2.py3-none-any.whl (689kB)
    100% |████████████████████████████████| 696kB 2.0MB/s 
Collecting six (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting singledispatch; python_version < "3.4" (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/c5/10/369f50bcd4621b263927b0a1519987a04383d4a98fb10438042ad410cf88/singledispatch-3.4.0.3-py2.py3-none-any.whl
Collecting astroid<2.0,>=1.6 (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/8b/29/0f7ec6fbf28a158886b7de49aee3a77a8a47a7e24c60e9fd6ec98ee2ec02/astroid-1.6.6-py2.py3-none-any.whl (305kB)
    100% |████████████████████████████████| 307kB 3.3MB/s 
Collecting configparser; python_version == "2.7" (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/7a/2a/95ed0501cf5d8709490b1d3a3f9b5cf340da6c433f896bbe9ce08dbe6785/configparser-4.0.2-py2.py3-none-any.whl
Collecting backports.functools-lru-cache; python_version == "2.7" (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/da/d1/080d2bb13773803648281a49e3918f65b31b7beebf009887a529357fd44a/backports.functools_lru_cache-1.6.1-py2.py3-none-any.whl
Collecting mccabe (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/87/89/479dc97e18549e21354893e4ee4ef36db1d237534982482c3681ee6e7b57/mccabe-0.6.1-py2.py3-none-any.whl
Collecting isort>=4.2.5 (from pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/e5/b0/c121fd1fa3419ea9bfd55c7f9c4fedfec5143208d8c7ad3ce3db6c623c21/isort-4.3.21-py2.py3-none-any.whl (42kB)
    100% |████████████████████████████████| 51kB 11.0MB/s 
Collecting wrapt (from astroid<2.0,>=1.6->pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/23/84/323c2415280bc4fc880ac5050dddfb3c8062c2552b34c2e512eb4aa68f79/wrapt-1.11.2.tar.gz
Collecting enum34>=1.1.3; python_version < "3.4" (from astroid<2.0,>=1.6->pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting lazy-object-proxy (from astroid<2.0,>=1.6->pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/9b/1c/a01cbc29f2f737a797e4abdd5a9436e52449b8f439fa0221806966c25530/lazy_object_proxy-1.4.3-cp27-cp27mu-manylinux1_x86_64.whl (54kB)
    100% |████████████████████████████████| 61kB 10.6MB/s 
Collecting futures; python_version < "3.2" (from isort>=4.2.5->pylint==1.9.4)
  Downloading https://files.pythonhosted.org/packages/d8/a6/f46ae3f1da0cd4361c344888f59ec2f5785e69c872e175a748ef6071cdb5/futures-3.3.0-py2-none-any.whl
Building wheels for collected packages: wrapt
  Running setup.py bdist_wheel for wrapt ... - \ done
  Stored in directory: /home/circleci/.cache/pip/wheels/d7/de/2e/efa132238792efb6459a96e85916ef8597fcb3d2ae51590dfd
Successfully built wrapt
Installing collected packages: six, singledispatch, wrapt, enum34, backports.functools-lru-cache, lazy-object-proxy, astroid, configparser, mccabe, futures, isort, pylint
Successfully installed astroid-1.6.6 backports.functools-lru-cache-1.6.1 configparser-4.0.2 enum34-1.1.6 futures-3.3.0 isort-4.3.21 lazy-object-proxy-1.4.3 mccabe-0.6.1 pylint-1.9.4 singledispatch-3.4.0.3 six-1.13.0 wrapt-1.11.2
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if Pillow is installed in /home/circleci/oppia/../oppia_tools
Installing Pillow
Checking if pip is installed on the local machine
Collecting Pillow==6.0.0
  Downloading https://files.pythonhosted.org/packages/b6/4b/5adc1109908266554fb978154c797c7d71aba43dd15508d8c1565648f6bc/Pillow-6.0.0-cp27-cp27mu-manylinux1_x86_64.whl (2.0MB)
    100% |████████████████████████████████| 2.0MB 671kB/s 
Installing collected packages: Pillow
Successfully installed Pillow-6.0.0
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if pylint-quotes is installed in /home/circleci/oppia/../oppia_tools
Installing pylint-quotes
Checking if pip is installed on the local machine
Collecting pylint-quotes==0.1.8
  Downloading https://files.pythonhosted.org/packages/28/f5/e00a9b7597ba7b92b58ddac472f9cf5bfd347022a65a4beafd5f682a7758/pylint_quotes-0.1.8-py2.py3-none-any.whl
Collecting pylint (from pylint-quotes==0.1.8)
  Downloading https://files.pythonhosted.org/packages/36/3b/fa4025a424adafd85c6195001b1c130ecb8d8b30784a1c4cb68e7b5e5ae7/pylint-1.9.5-py2.py3-none-any.whl (696kB)
    100% |████████████████████████████████| 696kB 1.9MB/s 
Collecting six (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting singledispatch; python_version < "3.4" (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/c5/10/369f50bcd4621b263927b0a1519987a04383d4a98fb10438042ad410cf88/singledispatch-3.4.0.3-py2.py3-none-any.whl
Collecting astroid<2.0,>=1.6 (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/8b/29/0f7ec6fbf28a158886b7de49aee3a77a8a47a7e24c60e9fd6ec98ee2ec02/astroid-1.6.6-py2.py3-none-any.whl
Collecting configparser; python_version == "2.7" (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/7a/2a/95ed0501cf5d8709490b1d3a3f9b5cf340da6c433f896bbe9ce08dbe6785/configparser-4.0.2-py2.py3-none-any.whl
Collecting backports.functools-lru-cache; python_version == "2.7" (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/da/d1/080d2bb13773803648281a49e3918f65b31b7beebf009887a529357fd44a/backports.functools_lru_cache-1.6.1-py2.py3-none-any.whl
Collecting mccabe (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/87/89/479dc97e18549e21354893e4ee4ef36db1d237534982482c3681ee6e7b57/mccabe-0.6.1-py2.py3-none-any.whl
Collecting isort>=4.2.5 (from pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/e5/b0/c121fd1fa3419ea9bfd55c7f9c4fedfec5143208d8c7ad3ce3db6c623c21/isort-4.3.21-py2.py3-none-any.whl
Collecting wrapt (from astroid<2.0,>=1.6->pylint->pylint-quotes==0.1.8)
Collecting enum34>=1.1.3; python_version < "3.4" (from astroid<2.0,>=1.6->pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting lazy-object-proxy (from astroid<2.0,>=1.6->pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/9b/1c/a01cbc29f2f737a797e4abdd5a9436e52449b8f439fa0221806966c25530/lazy_object_proxy-1.4.3-cp27-cp27mu-manylinux1_x86_64.whl
Collecting futures; python_version < "3.2" (from isort>=4.2.5->pylint->pylint-quotes==0.1.8)
  Using cached https://files.pythonhosted.org/packages/d8/a6/f46ae3f1da0cd4361c344888f59ec2f5785e69c872e175a748ef6071cdb5/futures-3.3.0-py2-none-any.whl
Installing collected packages: six, singledispatch, wrapt, enum34, backports.functools-lru-cache, lazy-object-proxy, astroid, configparser, mccabe, futures, isort, pylint, pylint-quotes
Successfully installed astroid-1.6.6 backports.functools-lru-cache-1.6.1 configparser-4.0.2 enum34-1.1.6 futures-3.3.0 isort-4.3.21 lazy-object-proxy-1.4.3 mccabe-0.6.1 pylint-1.9.5 pylint-quotes-0.1.8 singledispatch-3.4.0.3 six-1.13.0 wrapt-1.11.2
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if webtest is installed in /home/circleci/oppia/../oppia_tools
Installing webtest
Checking if pip is installed on the local machine
Collecting webtest==2.0.33
  Downloading https://files.pythonhosted.org/packages/47/fd/569d7e5ce6ea92d8ff6dd7a91f125762d88b199f2b4e17d7030a703f241d/WebTest-2.0.33-py2.py3-none-any.whl
Collecting six (from webtest==2.0.33)
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting WebOb>=1.2 (from webtest==2.0.33)
  Downloading https://files.pythonhosted.org/packages/06/e1/4acd2b4327fceb4c6446bdbca515f807ab83188526fd654940c00bcf8cc3/WebOb-1.8.5-py2.py3-none-any.whl (113kB)
    100% |████████████████████████████████| 122kB 9.1MB/s 
Collecting beautifulsoup4 (from webtest==2.0.33)
  Downloading https://files.pythonhosted.org/packages/f9/d9/183705a87492249b212d88eef740995f55076195bcf45ed59306c146e42d/beautifulsoup4-4.8.1-py2-none-any.whl (102kB)
    100% |████████████████████████████████| 102kB 9.3MB/s 
Collecting waitress>=0.8.5 (from webtest==2.0.33)
  Downloading https://files.pythonhosted.org/packages/2d/0b/84a7da7fd12ee556dd193c24b635a11b840d314bc580771dd2a3151450c9/waitress-1.3.1-py2.py3-none-any.whl (144kB)
    100% |████████████████████████████████| 153kB 7.5MB/s 
Collecting soupsieve>=1.2 (from beautifulsoup4->webtest==2.0.33)
  Downloading https://files.pythonhosted.org/packages/81/94/03c0f04471fc245d08d0a99f7946ac228ca98da4fa75796c507f61e688c2/soupsieve-1.9.5-py2.py3-none-any.whl
Collecting backports.functools-lru-cache; python_version < "3" (from soupsieve>=1.2->beautifulsoup4->webtest==2.0.33)
  Using cached https://files.pythonhosted.org/packages/da/d1/080d2bb13773803648281a49e3918f65b31b7beebf009887a529357fd44a/backports.functools_lru_cache-1.6.1-py2.py3-none-any.whl
Installing collected packages: six, WebOb, backports.functools-lru-cache, soupsieve, beautifulsoup4, waitress, webtest
Successfully installed WebOb-1.8.5 backports.functools-lru-cache-1.6.1 beautifulsoup4-4.8.1 six-1.13.0 soupsieve-1.9.5 waitress-1.3.1 webtest-2.0.33
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if isort is installed in /home/circleci/oppia/../oppia_tools
Installing isort
Checking if pip is installed on the local machine
Collecting isort==4.3.20
  Downloading https://files.pythonhosted.org/packages/1c/d9/bf5848b376e441ff358a14b954476423eeeb8c9b78c10074b7f53ce2918d/isort-4.3.20-py2.py3-none-any.whl (42kB)
    100% |████████████████████████████████| 51kB 11.1MB/s 
Collecting backports.functools-lru-cache; python_version < "3.2" (from isort==4.3.20)
  Using cached https://files.pythonhosted.org/packages/da/d1/080d2bb13773803648281a49e3918f65b31b7beebf009887a529357fd44a/backports.functools_lru_cache-1.6.1-py2.py3-none-any.whl
Collecting futures; python_version < "3.2" (from isort==4.3.20)
  Using cached https://files.pythonhosted.org/packages/d8/a6/f46ae3f1da0cd4361c344888f59ec2f5785e69c872e175a748ef6071cdb5/futures-3.3.0-py2-none-any.whl
Installing collected packages: backports.functools-lru-cache, futures, isort
Successfully installed backports.functools-lru-cache-1.6.1 futures-3.3.0 isort-4.3.20
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if pycodestyle is installed in /home/circleci/oppia/../oppia_tools
Installing pycodestyle
Checking if pip is installed on the local machine
Collecting pycodestyle==2.5.0
  Downloading https://files.pythonhosted.org/packages/0e/0c/04a353e104d2f324f8ee5f4b32012618c1c86dd79e52a433b64fceed511b/pycodestyle-2.5.0-py2.py3-none-any.whl (51kB)
    100% |████████████████████████████████| 51kB 7.7MB/s 
Installing collected packages: pycodestyle
Successfully installed pycodestyle-2.5.0
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if esprima is installed in /home/circleci/oppia/../oppia_tools
Installing esprima
Checking if pip is installed on the local machine
Collecting esprima==4.0.1
  Downloading https://files.pythonhosted.org/packages/cc/a1/50fccd68a12bcfc27adfc9969c090286670a9109a0259f3f70943390b721/esprima-4.0.1.tar.gz (47kB)
    100% |████████████████████████████████| 51kB 8.9MB/s 
Building wheels for collected packages: esprima
  Running setup.py bdist_wheel for esprima ... - done
  Stored in directory: /home/circleci/.cache/pip/wheels/63/aa/c8/3fbb4a3f263a33d0f89439d173d2b835426e4399a99f0c8e14
Successfully built esprima
Installing collected packages: esprima
Successfully installed esprima-4.0.1
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if browsermob-proxy is installed in /home/circleci/oppia/../oppia_tools
Installing browsermob-proxy
Checking if pip is installed on the local machine
Collecting browsermob-proxy==0.8.0
  Downloading https://files.pythonhosted.org/packages/f7/cc/6990849a17e1756b8b0173b0ad6cf0b3304ae2bfeaafb9e6f4038af176b3/browsermob_proxy-0.8.0-py2-none-any.whl
Collecting requests>=2.9.1 (from browsermob-proxy==0.8.0)
  Downloading https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl (57kB)
    100% |████████████████████████████████| 61kB 9.4MB/s 
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests>=2.9.1->browsermob-proxy==0.8.0)
  Downloading https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl (125kB)
    100% |████████████████████████████████| 133kB 9.3MB/s 
Collecting certifi>=2017.4.17 (from requests>=2.9.1->browsermob-proxy==0.8.0)
  Downloading https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl (156kB)
    100% |████████████████████████████████| 163kB 7.8MB/s 
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.9.1->browsermob-proxy==0.8.0)
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 8.8MB/s 
Collecting idna<2.9,>=2.5 (from requests>=2.9.1->browsermob-proxy==0.8.0)
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
    100% |████████████████████████████████| 61kB 10.1MB/s 
Installing collected packages: urllib3, certifi, chardet, idna, requests, browsermob-proxy
Successfully installed browsermob-proxy-0.8.0 certifi-2019.11.28 chardet-3.0.4 idna-2.8 requests-2.22.0 urllib3-1.25.7
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if selenium is installed in /home/circleci/oppia/../oppia_tools
Installing selenium
Checking if pip is installed on the local machine
Collecting selenium==3.13.0
  Downloading https://files.pythonhosted.org/packages/41/c6/78a9a0d0150dbf43095c6f422fdf6f948e18453c5ebbf92384175b372ca2/selenium-3.13.0-py2.py3-none-any.whl (946kB)
    100% |████████████████████████████████| 952kB 1.4MB/s 
Installing collected packages: selenium
Successfully installed selenium-3.13.0
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if PyGithub is installed in /home/circleci/oppia/../oppia_tools
Installing PyGithub
Checking if pip is installed on the local machine
Collecting PyGithub==1.43.7
  Downloading https://files.pythonhosted.org/packages/f7/48/ec5ef5239f3a4043ee7e07b454b5f3a024a5f22a4e585400ac6caa22c3c4/PyGithub-1.43.7.tar.gz (107kB)
    100% |████████████████████████████████| 112kB 9.7MB/s 
Collecting deprecated (from PyGithub==1.43.7)
  Downloading https://files.pythonhosted.org/packages/f6/89/62912e01f3cede11edcc0abf81298e3439d9c06c8dce644369380ed13f6d/Deprecated-1.2.7-py2.py3-none-any.whl
Collecting pyjwt (from PyGithub==1.43.7)
  Downloading https://files.pythonhosted.org/packages/87/8b/6a9f14b5f781697e51259d81657e6048fd31a113229cf346880bb7545565/PyJWT-1.7.1-py2.py3-none-any.whl
Collecting requests>=2.14.0 (from PyGithub==1.43.7)
  Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting wrapt<2,>=1.10 (from deprecated->PyGithub==1.43.7)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests>=2.14.0->PyGithub==1.43.7)
  Using cached https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.14.0->PyGithub==1.43.7)
  Using cached https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.14.0->PyGithub==1.43.7)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5 (from requests>=2.14.0->PyGithub==1.43.7)
  Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Building wheels for collected packages: PyGithub
  Running setup.py bdist_wheel for PyGithub ... - \ done
  Stored in directory: /home/circleci/.cache/pip/wheels/41/a2/13/f40c7a91db4b3743ef1830234526054d62f8c502510f3eb18a
Successfully built PyGithub
Installing collected packages: wrapt, deprecated, pyjwt, urllib3, certifi, chardet, idna, requests, PyGithub
Successfully installed PyGithub-1.43.7 certifi-2019.11.28 chardet-3.0.4 deprecated-1.2.7 idna-2.8 pyjwt-1.7.1 requests-2.22.0 urllib3-1.25.7 wrapt-1.11.2
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Checking if pygsheets is installed in /home/circleci/oppia/../oppia_tools
Installing pygsheets
Checking if pip is installed on the local machine
Collecting pygsheets==2.0.2
  Downloading https://files.pythonhosted.org/packages/3e/d1/29099dfa230c25667af84a4f0b7033aa3bcd0b36299bb829c78cd69b195b/pygsheets-2.0.2-py2-none-any.whl (97kB)
    100% |████████████████████████████████| 102kB 9.5MB/s 
Collecting google-api-python-client>=1.5.5 (from pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/31/c7/16ca16d28f2d71c8bd6fa67c91eb2a82259dc589c0504f903b675ecdaa84/google_api_python_client-1.7.11-py2-none-any.whl (55kB)
    100% |████████████████████████████████| 61kB 7.8MB/s 
Collecting google-auth-oauthlib (from pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/7b/b8/88def36e74bee9fce511c9519571f4e485e890093ab7442284f4ffaef60b/google_auth_oauthlib-0.4.1-py2.py3-none-any.whl
Collecting enum34 (from pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting six<2dev,>=1.6.1 (from google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting google-auth>=1.4.1 (from google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/b6/48/50934afa81c9ab6d68acc19c1c0543f765699fa4e3a36d675baa97816bb1/google_auth-1.8.1-py2.py3-none-any.whl (75kB)
    100% |████████████████████████████████| 81kB 11.5MB/s 
Collecting httplib2<1dev,>=0.9.2 (from google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/ce/2e/87461bfbb7e561203b759b3f7f639e2144226604372830d00a8279960ae1/httplib2-0.14.0.tar.gz (219kB)
    100% |████████████████████████████████| 225kB 5.0MB/s 
Collecting google-auth-httplib2>=0.0.3 (from google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/33/49/c814d6d438b823441552198f096fcd0377fd6c88714dbed34f1d3c8c4389/google_auth_httplib2-0.0.3-py2.py3-none-any.whl
Collecting uritemplate<4dev,>=3.0.0 (from google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/f6/25/66a49231b44409d7f07cfcf2506a8b070ce3c99fc47cc256bea833f24791/uritemplate-3.0.0-py2-none-any.whl
Collecting requests-oauthlib>=0.7.0 (from google-auth-oauthlib->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/a3/12/b92740d845ab62ea4edf04d2f4164d82532b5a0b03836d4d4e71c6f3d379/requests_oauthlib-1.3.0-py2.py3-none-any.whl
Collecting pyasn1-modules>=0.2.1 (from google-auth>=1.4.1->google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/52/50/bb4cefca37da63a0c52218ba2cb1b1c36110d84dcbae8aa48cd67c5e95c2/pyasn1_modules-0.2.7-py2.py3-none-any.whl (131kB)
    100% |████████████████████████████████| 133kB 9.4MB/s 
Collecting setuptools>=40.3.0 (from google-auth>=1.4.1->google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/54/28/c45d8b54c1339f9644b87663945e54a8503cfef59cf0f65b3ff5dd17cf64/setuptools-42.0.2-py2.py3-none-any.whl (583kB)
    100% |████████████████████████████████| 583kB 2.4MB/s 
Collecting cachetools<3.2,>=2.0.0 (from google-auth>=1.4.1->google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/2f/a6/30b0a0bef12283e83e58c1d6e7b5aabc7acfc4110df81a4471655d33e704/cachetools-3.1.1-py2.py3-none-any.whl
Collecting rsa<4.1,>=3.1.4 (from google-auth>=1.4.1->google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/02/e5/38518af393f7c214357079ce67a317307936896e961e35450b70fad2a9cf/rsa-4.0-py2.py3-none-any.whl
Collecting oauthlib>=3.0.0 (from requests-oauthlib>=0.7.0->google-auth-oauthlib->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/05/57/ce2e7a8fa7c0afb54a0581b14a65b56e62b5759dbc98e80627142b8a3704/oauthlib-3.1.0-py2.py3-none-any.whl (147kB)
    100% |████████████████████████████████| 153kB 8.0MB/s 
Collecting requests>=2.0.0 (from requests-oauthlib>=0.7.0->google-auth-oauthlib->pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting pyasn1<0.5.0,>=0.4.6 (from pyasn1-modules>=0.2.1->google-auth>=1.4.1->google-api-python-client>=1.5.5->pygsheets==2.0.2)
  Downloading https://files.pythonhosted.org/packages/62/1e/a94a8d635fa3ce4cfc7f506003548d0a2447ae76fd5ca53932970fe3053f/pyasn1-0.4.8-py2.py3-none-any.whl (77kB)
    100% |████████████████████████████████| 81kB 11.5MB/s 
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib->pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib->pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib->pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5 (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib->pygsheets==2.0.2)
  Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Building wheels for collected packages: httplib2
  Running setup.py bdist_wheel for httplib2 ... - done
  Stored in directory: /home/circleci/.cache/pip/wheels/51/ab/65/f318b4271019823cf7d7c178ac76a8ea4d6e6271fe9f5ba488
Successfully built httplib2
Installing collected packages: six, pyasn1, pyasn1-modules, setuptools, cachetools, rsa, google-auth, httplib2, google-auth-httplib2, uritemplate, google-api-python-client, oauthlib, urllib3, certifi, chardet, idna, requests, requests-oauthlib, google-auth-oauthlib, enum34, pygsheets
Successfully installed cachetools-3.1.1 certifi-2019.11.28 chardet-3.0.4 enum34-1.1.6 google-api-python-client-1.7.11 google-auth-1.8.1 google-auth-httplib2-0.0.3 google-auth-oauthlib-0.4.1 httplib2-0.14.0 idna-2.8 oauthlib-3.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.7 pygsheets-2.0.2 requests-2.22.0 requests-oauthlib-1.3.0 rsa-4.0 setuptools-42.0.2 six-1.13.0 uritemplate-3.0.0 urllib3-1.25.7
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Installing third-party JS libraries and zip files.
Downloading file yuicompressor-2.4.8.jar to ../oppia_tools/yuicompressor-2.4.8 ...
Download of yuicompressor-2.4.8.jar succeeded.
Downloading and unzipping file browsermob-proxy-2.1.1 to ../oppia_tools ...
Download of browsermob-proxy-2.1.1 succeeded.
Downloading and unzipping file ui-sortable-src0.19.0 to ./third_party/static ...
Download of ui-sortable-src0.19.0 succeeded.
Downloading and unzipping file MathJax-2.7.5 to ./third_party/static ...
Download of MathJax-2.7.5 succeeded.
Downloading and unzipping file ng-joyride-b117e019582433995393f81e2006a6bc36ca1cf2 to ./third_party/static ...
Download of ng-joyride-b117e019582433995393f81e2006a6bc36ca1cf2 succeeded.
Downloading and unzipping file bower-material-1.1.19 to ./third_party/static ...
Download of bower-material-1.1.19 succeeded.
Downloading and unzipping file angular-ui-tree-master to ./third_party/static ...
Download of angular-ui-tree-master succeeded.
Downloading and unzipping file fontawesome-free-5.9.0-web to ./third_party/static ...
Download of fontawesome-free-5.9.0-web succeeded.
Downloading file angular-translate-loader-static-files.min.js to ./third_party/static/bower-angular-translate-loader-static-files-2.18.1 ...
Download of angular-translate-loader-static-files.min.js succeeded.
Downloading and unzipping file guppy-b5055b963fdbea5c6c1e92dbf58fdaf3ea0cd8ba to ./third_party/static ...
Download of guppy-b5055b963fdbea5c6c1e92dbf58fdaf3ea0cd8ba succeeded.
Downloading file angular.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular.js succeeded.
Downloading file angular.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular.min.js succeeded.
Downloading file angular.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular.min.js.map succeeded.
Downloading file angular-animate.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-animate.js succeeded.
Downloading file angular-animate.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-animate.min.js succeeded.
Downloading file angular-animate.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular-animate.min.js.map succeeded.
Downloading file angular-resource.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-resource.js succeeded.
Downloading file angular-resource.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-resource.min.js succeeded.
Downloading file angular-resource.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular-resource.min.js.map succeeded.
Downloading file angular-route.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-route.js succeeded.
Downloading file angular-route.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-route.min.js succeeded.
Downloading file angular-route.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular-route.min.js.map succeeded.
Downloading file angular-sanitize.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-sanitize.js succeeded.
Downloading file angular-sanitize.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-sanitize.min.js succeeded.
Downloading file angular-sanitize.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular-sanitize.min.js.map succeeded.
Downloading file angular-aria.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-aria.js succeeded.
Downloading file angular-aria.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-aria.min.js succeeded.
Downloading file angular-aria.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular-aria.min.js.map succeeded.
Downloading file angular-touch.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-touch.js succeeded.
Downloading file angular-touch.min.js to ./third_party/static/angularjs-1.5.8 ...
Download of angular-touch.min.js succeeded.
Downloading file angular-touch.min.js.map to ./third_party/static/angularjs-1.5.8 ...
Download of angular-touch.min.js.map succeeded.
Downloading file ui-bootstrap-tpls-2.5.0.js to ./third_party/static/ui-bootstrap-2.5.0 ...
Download of ui-bootstrap-tpls-2.5.0.js succeeded.
Downloading file ui-bootstrap-tpls-2.5.0.min.js to ./third_party/static/ui-bootstrap-2.5.0 ...
Download of ui-bootstrap-tpls-2.5.0.min.js succeeded.
Downloading and unzipping file jquery-ui-touch-punch-improved-0.3.1 to ./third_party/static ...
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/local/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/circleci/oppia/scripts/install_third_party_libs.py", line 312, in <module>
    main()
  File "/home/circleci/oppia/scripts/install_third_party_libs.py", line 287, in main
    install_third_party.main(args=[])
  File "scripts/install_third_party.py", line 332, in main
    download_manifest_files(MANIFEST_FILE_PATH)
  File "scripts/install_third_party.py", line 317, in download_manifest_files
    dependency_zip_root_name, dependency_target_root_name)
  File "scripts/install_third_party.py", line 127, in download_and_unzip_files
    python_utils.url_retrieve(source_url, filename=TMP_UNZIP_PATH)
  File "python_utils.py", line 289, in url_retrieve
    return urllib.urlretrieve(source_url, filename=filename)
  File "/usr/local/lib/python2.7/urllib.py", line 98, in urlretrieve
    return opener.retrieve(url, filename, reporthook, data)
  File "/usr/local/lib/python2.7/urllib.py", line 245, in retrieve
    fp = self.open(url, data)
  File "/usr/local/lib/python2.7/urllib.py", line 213, in open
    return getattr(self, name)(url)
  File "/usr/local/lib/python2.7/urllib.py", line 443, in open_https
    h.endheaders(data)
  File "/usr/local/lib/python2.7/httplib.py", line 1038, in endheaders
    self._send_output(message_body)
  File "/usr/local/lib/python2.7/httplib.py", line 882, in _send_output
    self.send(msg)
  File "/usr/local/lib/python2.7/httplib.py", line 844, in send
    self.connect()
  File "/usr/local/lib/python2.7/httplib.py", line 1255, in connect
    HTTPConnection.connect(self)
  File "/usr/local/lib/python2.7/httplib.py", line 821, in connect
    self.timeout, self.source_address)
  File "/usr/local/lib/python2.7/socket.py", line 575, in create_connection
    raise err
IOError: [Errno socket error] [Errno 110] Connection timed out

Exited with code exit status 1
</details>

--- a/core/templates/dev/head/services/UpgradedServices.ts
+++ b/core/templates/dev/head/services/UpgradedServices.ts
@@ -317,7 +317,6 @@ export class UpgradedServices {
       new WrittenTranslationsObjectFactory(
         upgradedServices['WrittenTranslationObjectFactory']);
 
-
     // Group 3: Services depending only on groups 1-2.
     upgradedServices['AnswerGroupObjectFactory'] =
       new AnswerGroupObjectFactory(

